src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.gradle.plugins.jpi
+
+import org.gradle.api.artifacts.CacheableRule
+import org.gradle.api.artifacts.ComponentMetadataContext
+import org.gradle.api.artifacts.ComponentMetadataRule
+import org.gradle.api.attributes.LibraryElements
+import org.gradle.api.model.ObjectFactory
+
+import javax.inject.Inject
+
+@CacheableRule
+class JpiVariantRule implements ComponentMetadataRule {
+
+    @Inject
+    ObjectFactory getObjects() {
```

Comment:
Can you make the getter `abstract`?

Reply:
Indeed. I was not aware it was working for rules. I have to update some other examples. :) Thanks.